src/courses/services/course.service.ts

```diff
@@ -88,14 +88,14 @@ export class CourseService extends BaseService<Course> {
 
   async updateJoinedUsers(id: string, user: ClientUser, action: ActionCourse) {
     const course = await this.findById(id, {relations: ["joinedUsers"]});
-    const users = !course.joinedUsers ? [] : course.joinedUsers;
-    const checkAvaiable = _.filter(users, ['id', user.id]);
+    const users: ClientUser[] = _.clone(course.joinedUsers);
+    const checkAvailable = _.some(users, ['id', user.id]);
 
     if(action === ActionCourse.JOIN) {
-      if(checkAvaiable.length !== 0) return false;
+      if(checkAvailable) { return false };
       users.push(user)
     }else{
-      if(checkAvaiable.length === 0) return false;
+      if(!checkAvailable) { return false };
       _.remove(users, (oldUser) => oldUser.id === user.id)
     }

```

Comment:
A đề xuất thuật toán chỗ này, e có thể cách ra làm 2 hàm (và gộp vào cũng được) và truyền vào như sau 
`addJoinedUser(clientUser: ClientUser)` => dùng để add joined user
`removeJoinedUser(clientUser: ClientUser)` => dùng để remove joined user

Bởi vì nếu bên achievement đã có thể add/remove được hoặc return true rồi thì bên này chắc chắn cũng có thể add user vào được, e không cần check available ở đây nữa. Tối giản logic ở đây lại